posthog/api/test/test_event.py

```diff
@@ -686,3 +686,694 @@ def test_filter_events_with_date_format(self):
+
+        cohort1 = Cohort.objects.create(
+            team=self.team,
+            name="cohort_1",
+            groups=[{"properties": [{"key": "key", "value": "value", "type": "person"}]}],
+        )
+
+        cohort1.calculate_people_ch(pending_version=0)
+
+        with self.settings(USE_PRECALCULATED_CH_COHORT_PEOPLE=True):  # Normally this is False in tests
+            with freeze_time("2020-01-04T13:01:01Z"):
+                response = self.client.get(
+                    f"/api/projects/{self.team.id}/events/?properties=%s"
+                    % (json.dumps([{"key": "id", "value": cohort1.id, "type": "cohort"}]))
+                ).json()
+
+        self.assertEqual(len(response["results"]), 2)
+
+    @patch("posthoganalytics.feature_enabled", return_value=True)
+    def test_filter_by_person(self, _):
+        person = _create_person(
+            properties={"email": "tim@posthog.com"},
+            distinct_ids=["2", "some-random-uid"],
+            team=self.team,
+            immediate=True,
+        )
+
+        _create_event(event="random event", team=self.team, distinct_id="2", properties={"$ip": "8.8.8.8"})
+        _create_event(
+            event="random event", team=self.team, distinct_id="some-random-uid", properties={"$ip": "8.8.8.8"}
+        )
+        _create_event(event="random event", team=self.team, distinct_id="some-other-one", properties={"$ip": "8.8.8.8"})
+        flush_persons_and_events()
+
+        response = self.client.get(f"/api/projects/{self.team.id}/events/?person_id={person.pk}").json()
+        self.assertEqual(len(response["results"]), 2)
+        self.assertEqual(response["results"][0]["elements"], [])
+
+        response = self.client.get(f"/api/projects/{self.team.id}/events/?person_id={person.uuid}").json()
+        self.assertEqual(len(response["results"]), 2)
+
+    @patch("posthoganalytics.feature_enabled", return_value=True)
+    def test_filter_by_nonexisting_person(self, _):
+        response = self.client.get(f"/api/projects/{self.team.id}/events/?person_id=5555555555")
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(len(response.json()["results"]), 0)
+
+    @patch("posthoganalytics.feature_enabled", return_value=True)
+    def test_custom_event_values(self, _):
+        events = ["test", "new event", "another event"]
+        for event in events:
+            _create_event(
+                distinct_id="bla",
+                event=event,
+                team=self.team,
+                properties={"random_prop": "don't include", "some other prop": "with some text"},
+            )
+        response = self.client.get(f"/api/projects/{self.team.id}/events/values/?key=custom_event").json()
+        self.assertListEqual(sorted(events), sorted(event["name"] for event in response))
+
+    @patch("posthoganalytics.feature_enabled", return_value=True)
+    @test_with_materialized_columns(["random_prop"])
```

Comment:
How do materialized columns work with the live events table? Presumable the events and live_events tables need to be in sync to be able to generate a valid query?